pkg/controller/kabaneroplatform/stack-controller.go

```diff
@@ -219,3 +298,75 @@ func getStackControllerStatus(ctx context.Context, k *kabanerov1alpha2.Kabanero,
 
 	return ready, err
 }
+
+
+func pipelinesNamespace(k *kabanerov1alpha2.Kabanero) string {
+	var pipelinesNamespace string
+	if len(k.Spec.PipelinesNamespace) != 0 {
+		pipelinesNamespace = k.Spec.PipelinesNamespace
+	} else {
+		pipelinesNamespace = k.GetNamespace()
+	}
+	return pipelinesNamespace
+}
+
+
+// Clean up the old namespace SA/Role/Bindings
+func reconcilePipelinesNamespace(ctx context.Context, k *kabanerov1alpha2.Kabanero, c client.Client, _ logr.Logger) error {
+	logger := sclog.WithValues("Kabanero instance namespace", k.Namespace, "Kabanero instance Name", k.Name)
+	logger.Info("Reconciling Kabanero pipelinesNamespace.")
+
+	pipelinesNamespace := pipelinesNamespace(k)
+
+	deployedStacks := &kabanerov1alpha2.StackList{}
+	err := c.List(ctx, deployedStacks, client.InNamespace(k.GetNamespace()))
+	if err != nil {
+		return err
+	}
+
+	// Remove the old SA/Role/Bindings once all the Featured Stacks are re-created in the new pipelinesNamespace
+	canCleanup := true
+	for _, deployedStack := range deployedStacks.Items {
+		if k.Spec.PipelinesNamespace != deployedStack.Status.PipelinesNamespace {
+			canCleanup = false
+		}
+	}
+	
+	if canCleanup == true {
```

Comment:
I see what you are doing here.... I had not considered this before.  Yes we need to leave permissions around long enough for the stack controller to clean up the pipelines, etc, in the old namespace.  However when I tried switching the namespace on my cluster, this did not behave as I expected.
Old pipelinesNamespace was `fred`, new was `kabanero`.  First time thru, it created stuff in `kabanero` and I could see things transitioning to `kabanero`.  But then the kabanero controller reconcile was driven again, and this time it deleted all of the roles and bindings from `kabanero`.  I think this code needs to be able to detect that there is cleanup work to do for a namespace, and then when it's done, put a mark on the wall or something so that it doesn't try to do it again.